interfaces/apparmor/backend.go

```diff
@@ -152,6 +152,19 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 			logger.Noticef("cannot create host snap-confine apparmor configuration: %s", err)
 		}
 	}
+	// core on core devices is also special, the apparmor cache gets
+	// confused too easy, especially at rollbacks, so we delete the cache.
+	// See LP:#1460152 and
+	// https://forum.snapcraft.io/t/core-snap-revert-issues-on-core-devices/
+	if snapName == "core" && !release.OnClassic {
```

Comment:
I wonder if it would make sense check if the snap type is "os" instead of checking the snapName (I can see why you picked snapName-- the conditional immediately before does the same thing). I realize we are much farther out for non-Ubuntu Core devices with alternate os snap than alternate base snaps, so I won't block on that, but mentioning it for your consideration.